src/globalvars.cpp

```diff
@@ -96,19 +106,30 @@ void update_global_vars_rodata(
 
   // Update the values for the global vars (using the above offsets)
   for (const auto &[global_var, offset] : vars_and_offsets) {
-    auto *var = reinterpret_cast<int64_t *>(global_vars_buf + offset);
-
-    auto it = known_global_var_values.find(global_var);
-    if (it != known_global_var_values.end()) {
-      *var = it->second;
+    auto it = global_var_vals.find(global_var);
+    if (it != global_var_vals.end()) {
+      const auto &config = added_global_vars.at(global_var);
+      if (config.type == Type::integer) {
+        auto *var = reinterpret_cast<int64_t *>(global_vars_buf + offset);
+        *var = std::get<int64_t>(it->second);
+      } else if (config.type == Type::boolean) {
+        auto *var = reinterpret_cast<int8_t *>(global_vars_buf + offset);
+        *var = std::get<bool>(it->second) ? 1 : 0;
+      } else if (config.type == Type::string) {
+        auto *var = reinterpret_cast<char *>(global_vars_buf + offset);
+        const auto &val = std::get<std::string>(it->second);
+        strncpy(var, val.data(), val.size());
```

Comment:
You probably want `val.c_str()` and `val.size() + 1` to ensure that it is null-terminated, no? I'm not sure how string globals are sized, but just for safety and to ensure that things like `strlen` would give you the intended results (if you call this on the value in the kernel for some reason).